Client/Nimbus/NimbusFeatureFlagLayer.swift

```diff
@@ -187,6 +189,17 @@ final class NimbusFeatureFlagLayer {
             }
     }
 
+    public func checkNimbusForRustSyncManager(
```

Comment:
nit: Since you don't have access to FeatureFlaggable you can clean this up by removing it

Reply:
Putting this here for any other reviewers, the switch statement that `rustSyncManagerStatus` was added to needs to be exhaustive so keeping this code for now.